src/AspNet.Security.OAuth.SuperOffice/SuperOfficeAuthenticationHandler.cs

```diff
@@ -0,0 +1,196 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.JsonWebTokens;
+using Microsoft.IdentityModel.Tokens;
+
+namespace AspNet.Security.OAuth.SuperOffice
+{
+    /// <summary>
+    /// Defines a handler for authentication using SuperOffice.
+    /// </summary>
+    public class SuperOfficeAuthenticationHandler : OAuthHandler<SuperOfficeAuthenticationOptions>
+    {
+        public SuperOfficeAuthenticationHandler(
+            [NotNull] IOptionsMonitor<SuperOfficeAuthenticationOptions> options,
+            [NotNull] ILoggerFactory logger,
+            [NotNull] UrlEncoder encoder,
+            [NotNull] ISystemClock clock)
+            : base(options, logger, encoder, clock)
+        {
+        }
+
+        /// <inheritdoc />
+        protected override async Task<AuthenticationTicket> CreateTicketAsync(
+            [NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties,
+            [NotNull] OAuthTokenResponse tokens)
+        {
+            string contextId = await ProcessIdTokenAndGetContactIdentifierAsync(tokens, properties, identity);
```

Comment:
nit: we tend to use `var` everywhere, to be consistent with the ASP.NET code style.

Reply:
OK, I thought this was from your coding standards file... 

![image](https://user-images.githubusercontent.com/1012665/82228434-670f8280-9929-11ea-9d42-98cf0c89bd26.png)

Reply:
It does. It's up to Kévin to decide which wins, but I'm in the _"`var` is lazy, don't use it when the type isn't obvious"_ side of the debate, which is why the EditorConfig file is set up that way.

Reply:
Personally, I have no strong opinion: I've worked on projects using `var` everywhere, only with constructors or never, and it's really a matter of personal preference.

I think we should be consistent with the code style used in the MSFT providers.

Reply:
I'm a var guy too. I've change them to be vars.